src/routes/index.ts

```diff
@@ -271,8 +271,10 @@ router.post(config.UPDATE_CHECK_YOUR_ANSWERS_URL, authentication, updateCheckYou
 
 router.get(config.OVERSEAS_ENTITY_PAYMENT_WITH_TRANSACTION_URL, authentication, overseasEntityPayment.get);
 
-router.get(config.OVERSEAS_ENTITY_UPDATE_DETAILS_URL, authentication, overseasEntityUpdateDetails.get);
-router.post(config.OVERSEAS_ENTITY_UPDATE_DETAILS_URL, authentication, ...validator.entity, checkValidations, overseasEntityUpdateDetails.post);
+router.route(config.OVERSEAS_ENTITY_UPDATE_DETAILS_URL)
+  .all(authentication)
+  .get(overseasEntityUpdateDetails.get)
+  .post(...validator.entity, ...validator.overseasName, checkValidations, overseasEntityUpdateDetails.post);
```

Comment:
validator.overseasName is probably what is used in registration. We should be checking that we have the entityNumber. We may have the name too but we should re-direct back to the OE number query page if there is no OE number (say you browse directly to the details page URL).

Reply:
Sorry, the above comment is unclear. I was talking about the lack of navigation checks and that the entityName is probably used in registration but that new validation (of entity number) is needed. We failed to add the navigation and can be added here as already raised as know issue in response to my comment.

Reply:
Navigation check for OE number added

Reply:
checkValidations is not the right validation check. I'm expecting validation.hasEntityNumber or validation.hasEntity

Reply:
OE number check added. There is a unit test already existing for this navigation check.